### PR TITLE
Documenting the Documentation

### PR DESCRIPTION
**Pull request by jpbernius**
- Create docs/README.md
  Add a stating guide on how documentation works in this project and how to
  setup Sphinx locally.
- Update Sphinx
- Add sphinx-autobuild

### CODE REVIEW

File: docs/make.bat [hunk 1]
```diff
@@ -28,6 +29,9 @@ if errorlevel 9009 (
 %SPHINXBUILD% -M %1 %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
 goto end
 
+:livehtml
+sphinx-autobuild %SOURCEDIR% %BUILDDIR% %SPHINXOPTS% %O%
+
```
Comment (FrankeLukas) [inline docs/make.bat:34]: This needs a `goto end`, otherwise it'll continue into the `:help` section which I don't think is intentional.

**Review by FrankeLukas — COMMENTED** [review]
Lgtm otherwise.

**Comment by anditurdiu** [comment]
On windows, I had to install the sphinx_rtd_theme manually, `pip install sphinx_rtd_theme`

File: docs/make.bat (hunk 1 above)
Comment (jpbernius) [reply docs/make.bat:35]: Thanks, missed that one :+1:

**Review by krusche — APPROVED** [review]

**Comment by krusche** [comment]
Thanks! This looks really good to me. I'll merge this for now so that others who write documentation can profit.
In case we need to change something, please do this in a follow-up PR.

**Pull request merged**